signer/signer_trust.go

```diff
@@ -85,3 +96,41 @@ func (trust *NotarySigner) GetKey(keyid string) data.PublicKey {
 	}
 	return data.NewPublicKey(data.KeyAlgorithm(publicKey.KeyInfo.Algorithm.Algorithm), publicKey.PublicKey)
 }
+
+// CheckHealth returns true if trust is healthy - that is if it can connect
+// to the trust server, and both the key management and signer services are
+// healthy
+func (trust *NotarySigner) CheckHealth(timeout time.Duration) error {
```

Comment:
I think in this particular concrete implementation, as we know that the Key Manager and Signer are the same gRPC service, we probably don't need to check both, a check on one implies the other is up and available. Let me know if I'm missing some benefit of checking them separately.